erigon-lib/commitment/metrics.go

```diff
@@ -0,0 +1,331 @@
+package commitment
+
+import (
+	"encoding/csv"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"time"
+)
+
+/*
+ERIGON_COMMITMENT_TRACE - file path to write commitment metrics
+ERIGON_COMMITMENT_ACCOUNT_TRACE - file path to write account metrics
+*/
+func init() {
+	commitmentMetricsFile = os.Getenv("ERIGON_COMMITMENT_TRACE")
+	accountMetricsFile = os.Getenv("ERIGON_COMMITMENT_ACCOUNT_TRACE")
+	collectCommitmentMetrics = os.Getenv("ERIGON_COMMITMENT_TRACE") != ""
+	collectAccountMetrics = os.Getenv("ERIGON_COMMITMENT_ACCOUNT_TRACE") != ""
+}
+
+var (
+	commitmentMetricsFile    string
+	accountMetricsFile       string
+	collectCommitmentMetrics bool
+	collectAccountMetrics    bool
+)
+
+type Metrics interface {
+	Reset()
+	Headers() []string
+	Values() [][]string
+}
+
+type ProcessCommitment struct {
+	Updates             atomic.Uint64
+	AddressKeys         atomic.Uint64
+	StorageKeys         atomic.Uint64
+	LoadBranch          atomic.Uint64
+	LoadAccount         atomic.Uint64
+	LoadStorage         atomic.Uint64
+	UpdateBranch        atomic.Uint64
+	Unfolds             atomic.Uint64
+	TotalUnfoldingTime  time.Duration
+	TotalFoldingTime    time.Duration
+	TotalProcessingTime time.Duration
```

Comment:
could have both currentPlainKey and processAccount inside this struct so no double lines.

Since you made it non-pointer inside hph, i would assume add Enabled field which is TRUE only when such tracing is requested

in each method add if !enabled { return } straight away.